osf/metadata/serializers.py

```diff
@@ -0,0 +1,152 @@
+        return jsonschema.validate(cls.serialize(record), record.schema.schema)
+
+
+@register(schema_id='datacite')
+class DataciteMetadataRecordSerializer(MetadataRecordSerializer):
+
+    osf_schema = 'user_entered_datacite.json'
+
+    @classmethod
+    def serialize_json(cls, record):
+        osfstorage_file = record.file
+        target = osfstorage_file.target
+        target.reload()
+        doc = {
+            'creators': utils.datacite_format_contributors(target.visible_contributors),
+            'titles': [
+                {
+                    'title': osfstorage_file.name
+                },
+                {
+                    'title': target.title,
+                    'titleType': 'AlternativeTitle'
+                }
+            ],
+            'publisher': 'Open Science Framework',
+            'dates': [
+                {
+                    'date': str(osfstorage_file.created),
+                    'dateType': 'Created'
+                },
+                {
+                    'date': str(osfstorage_file.modified),
+                    'dateType': 'Updated'
+                }
+            ],
+        }
+
+        file_description = record.metadata.get('file_description')
+        if file_description:
+            doc['descriptions'] = [
+                {
+                    'description': file_description,
+                    'descriptionType': 'Abstract'
+                }
+            ]
+
+        subject_list = []
+        subjects_from_target = target.subjects.all().select_related('bepress_subject')
+        if subjects_from_target.exists():
+            subject_list = utils.datacite_format_subjects(subjects_from_target)
+        tags_on_file = osfstorage_file.tags.values_list('name', flat=True)
+        for tag_name in tags_on_file:
+            subject_list.append({'subject': tag_name})
+        if subject_list:
+            doc['subjects'] = subject_list
+
+        resource_type = record.metadata.get('resource_type', '(:unas)')
+        doc['resourceType'] = {
+            'resourceType': resource_type,
+            'resourceTypeGeneral': utils.RESOURCE_TYPE_MAP.get(resource_type)
+        }
+
+        doc['publicationYear'] = str(osfstorage_file.created.year)
+
+        related_publication_doi = record.metadata.get('related_publication_doi')
+        if related_publication_doi:
+            doc['relatedIdentifiers'] = [
+                {
+                    'relatedIdentifier': related_publication_doi,
+                    'relatedIdentifierType': 'DOI',
+                    'relationType': 'IsSupplementTo'
+                }
+            ]
+
+        if osfstorage_file.guids.exists():
+            doc['alternateIdentifiers'] = [
+                {
+                    'alternateIdentifier': DOMAIN + osfstorage_file.guids.first()._id,
+                    'alternateIdentifierType': 'URL'
+                }
+            ]
+
+        funder_name = record.metadata.get('funding_agency')
```

Comment:
We need to update this since we allow for a list of `funders`: https://github.com/CenterForOpenScience/osf.io/pull/8721/files#diff-06bf61b1aa6656fdef57a942f33507dbR39